qt build failure on High Sierra

I'm opening a separate issue from #14418 to track Qt build failures…

1. [QTBUG-62266](https://bugreports.qt.io/browse/QTBUG-62266?jql=text%20~%20%22macOS%2010.13%22)
2. [`CBCentralManagerState` is deprecated](https://codereview.qt-project.org/#/c/172072/), which I reported as [QTBUG-62658](https://bugreports.qt.io/browse/QTBUG-62658)
3. Now, I'm seeing a widely reported [missing header](https://forums.developer.apple.com/message/250753) which is apparently specific to Xcode 9 beta 5:

```
/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk/System/Library/Frameworks/AVFoundation.framework/Headers/AVFoundation.h:88:9: fatal error: 'AVFoundation/AVRouteDetector.h' file not found
#import <AVFoundation/AVRouteDetector.h>
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[411/18453] ACTION //cc/ipc:interfaces_shared__generator(//build/toolchain/mac:clang_x64)
[412/18453] CXX obj/ppapi/cpp/objects/precompile.h-cc.gch
[413/18453] CXX obj/ppapi/thunk/thunk/precompile.h-cc.gch
[414/18453] ACTION //cc/ipc:interfaces_shared__generator(//build/toolchain/mac:clang_x64)
[415/18453] ACTION //cc/ipc:interfaces_shared__generator(//build/toolchain/mac:clang_x64)
```

I have a [patch](https://github.com/Homebrew/formula-patches/blob/master/qt/high-sierra.diff) for the first two issues. Tomorrow I will install the next Xcode beta and see if the third issue is gone.

--- a/Formula/qt.rb
+++ b/Formula/qt.rb
@@ -58,6 +58,15 @@ class Qt < Formula
     sha256 "b18e4715fcef2992f051790d3784a54900508c93350c25b0f2228cb058567142"
   end
 
+  # Patch fixing bugs QTBUG-62266 and QTBUG-62658 on macOS 10.13 High Sierra
+  # https://github.com/Homebrew/homebrew-core/issues/17075
+  if MacOS.version >= :high_sierra
+    patch do
+      url "https://raw.githubusercontent.com/Homebrew/formula-patches/45282b5b48/qt/high-sierra.diff"
+      sha256 "d8589d747a9ce0b7b7ddf1b59c4d999bbf8a02261e047a602cff39bea151eb42"
+    end
+  end
+
   def install
     args = %W[
       -verbose

